src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1279,8 +1333,77 @@ private SourceColumnMetadata GetColumnMetadata(int ordinal)
                     method = ValueMethod.GetValue;
                 }
             }
+            else if (_rowSourceType == ValueSourceType.IDataReader)
+            {
+                isSqlType = false;
+                isDataFeed = false;
+
+                Type t = null;
+                try
+                {
+                    t = ((IDataReader)_rowSource).GetFieldType(ordinal);
+                }
+                catch
+                {
+                    // do some error logging here??
```

Comment:
What type of exception do you expect to catch here?

Reply:
I am attempting to code defensively to not break existing custom IDataReader implementations that could potentially throw an exception on `GetFieldType`

Any feedback here would be appreciated!

Reply:
The only risk here is the `ordinal` that it throws exception earlier.
I don't think this be necessary here. Also, this method has covered in a try-catch statement.

Reply:
okay sounds good